packages/uicore/src/components/ExpressionDropdown/ExpressionDropdown.tsx

```diff
@@ -58,10 +58,13 @@ export const NewExpressionDropdown = (props: NewExpressionDropdownProps): JSX.El
     return listItems
   }, [rootTrieNode, query])
 
-  const dropDownItemClickHandler = (value: string): void => {
-    const valueBefore = query.substring(0, query.lastIndexOf('<'))
-    setQueryValue(`${valueBefore}<+${value}`)
-  }
+  const dropDownItemClickHandler = React.useCallback(
+    (value: string): void => {
+      const valueBefore = query.substring(0, query.lastIndexOf('<'))
+      setQueryValue(`${valueBefore}<+${value}`)
+    },
+    [query]
+  )
```

Comment:
Can we check if a functional state update be used for `setQueryValue` to access the previous query and remove `query` from deps?